sinks/interamap/src/real_time_parser.py

```diff
@@ -0,0 +1,136 @@
+    lat = convert_to_decimal_degrees(latitude)
+    lon = convert_to_decimal_degrees(longitude)
+
+    if boardId == "PROCESSOR":
+        # If the board ID is PROCESSOR, the timestamp is not available
+        timestamp = None
+        lon, lat = switch_numbers_keep_sign(lon, lat) # NOTE: This is for fix a mistake in the PROCESSOR data
+        
+    # Combine altitude and decimal altitude
+    he = altitude["altitude"] + altitude["daltitude"] / 100
+
+    point = Point_GPS(lon=lon, lat=lat, he=he, num_sats=num_sats, time_stamp=timestamp, board_id=boardId)
+
+    return point
+
+def parse_gps_info(data):
+    info = data.get("data",{"num_sats": 0, "quality": 0})
+    num_sats = info["num_sats"]
+    quality = info["quality"]
+    boardId = data.get("board_id")
+
+    info = Info_GPS(num_sats=num_sats, quality=quality, board_id=boardId)
+
+    return info
+
+def switch_numbers_keep_sign(a, b):
+    # Swap absolute values without changing signs
+    new_a = abs(b) if a >= 0 else -abs(b)
+    new_b = abs(a) if b >= 0 else -abs(a)
+    return new_a, new_b
+
+def convert_to_decimal_degrees(coord):
+    degs = coord["degs"]
+    mins = coord["mins"] / 60
+    dmins = coord["dmins"] / 3600
+    decimal = degs + mins + dmins
+    if coord.get("direction") in ["S", "W"]:
+        decimal = -decimal
+    return decimal
+
+class RTParser(QThread):
+    gps_RT_data = Signal(object)
+
+    def __init__(self):
+        QThread.__init__(self)
+
+        self.receiver = Receiver("")
+        self.running = False
+
+    def run(self):
+        self.running = True
+        self.extract_gps_data()
+
+    def stop(self):
+        self.running = False
+
+    def __del__(self):
+        self.stop()
+
+    def extract_gps_data(self):
+        boardIds = ["GPS", "PROCESSOR"] # List of board IDs
+        gps = [{} for _ in boardIds]
+
+        while True:
+            # If stop() was called
+            if not self.running:
+                break
+
+            try:
+                data = self.receiver.recv()
+                msgtype = data.get("msg_type")
+                
+                if msgtype in ["GPS_INFO", "GPS_TIMESTAMP", "GPS_LATITUDE", "GPS_LONGITUDE", "GPS_ALTITUDE"]:
+                    board_id = data.get("board_id")
+                    gps[boardIds.index(board_id)][msgtype] = data["data"]
+                
+                if all(key in gps[0] for key in ["GPS_INFO", "GPS_TIMESTAMP", "GPS_LATITUDE", "GPS_LONGITUDE", "GPS_ALTITUDE"]): # GPS 
+                    point = parse_gps_data(gps[0], data)
+                    self.gps_RT_data.emit(point)
+                    gps[0].clear()
+                    
+                if all(key in gps[1] for key in ["GPS_INFO", "GPS_LATITUDE", "GPS_LONGITUDE", "GPS_ALTITUDE"]): # PROCESSOR (does not have GPS_TIMESTAMP)
+                    point = parse_gps_data(gps[1], data)
+                    self.gps_RT_data.emit(point)
+                    gps[1].clear()
```

Comment:
My main concern with this is that if a message was dropped, the different packets would get mixed together. Maybe we can do something with message order?

For example GPS board always send in the order of time-latitude-longitude-info-altitude. And processor always sends in order of longitude-latitude-altitude-info. So if we see the message is out of order we know it's a new packet already. Not liking it too much though because everytime order changes it breaks.

Maybe we can group it by message time stamp?

Reply:
That's a good idea :)) I will deep checking a bit for this.

Reply:
I did update based on your last commit, I want to merge this PR first since we also need to develop UI part something, but we can keep this in mind and later to come up with this suggestion.

Reply:
But I think that's a good idea to throw useless information but not keep useless things.

Reply:
ok sounds good, not need to block other stuff on this